src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -467,7 +467,7 @@ internal bool IsReferenceTypeFromConstraintTypes(ImmutableArray<TypeSymbolWithAn
 
         internal static bool? IsNotNullableIfReferenceTypeFromConstraintType(TypeSymbolWithAnnotations constraintType)
         {
-            if (constraintType.IsAnnotated)
+            if (constraintType.NullableAnnotation.IsAnyNullable())
```

Comment:
>.IsAnyNullable() [](start = 49, length = 16)

In some places (such as `IsValidConstraintType`), you chose to test `== NullableAnnotation.Nullable` rather than test both nullable values. Is there a reason?
It feels like we wouldn't get a `NullableBasedOnAnalysis` here, or am I missing something?